### PR TITLE
Do not panic on setting Bottlerocket AMI  with create nodegroup

### PR DESCRIPTION
**Pull request by faraazkhan**
### Description
resolves https://github.com/weaveworks/eksctl/issues/5697

Output from locally built binary (does not panic due to bad casing):
```
./eksctl create nodegroup --cluster multitenant --managed --node-ami-family BOTTLERoCkeT
2022-09-26 17:57:55 [ℹ]  will use version 1.22 for new nodegroup(s) based on control plane version
2022-09-26 17:57:57 [ℹ]  nodegroup "ng-e4cc9867" will use "" [Bottlerocket/1.22]
2022-09-26 17:58:04 [ℹ]  3 existing nodegroup(s) (ng-85cbae68,ng-945c938a,private) will be excluded
2022-09-26 17:58:04 [ℹ]  1 nodegroup (ng-e4cc9867) was included (based on the include/exclude rules)
2022-09-26 17:58:04 [ℹ]  will create a CloudFormation stack for each of 1 managed nodegroups in cluster "multitenant"
2022-09-26 17:58:04 [ℹ]
2 sequential tasks: { fix cluster compatibility, 1 task: { 1 task: { create managed nodegroup "ng-e4cc9867" } }
}
2022-09-26 17:58:04 [ℹ]  checking cluster stack for missing resources
2022-09-26 17:58:07 [ℹ]  cluster stack has all required resources
2022-09-26 17:58:11 [ℹ]  building managed nodegroup stack "eksctl-multitenant-nodegroup-ng-e4cc9867"
2022-09-26 17:58:12 [ℹ]  deploying stack "eksctl-multitenant-nodegroup-ng-e4cc9867"
2022-09-26 17:58:12 [ℹ]  waiting for CloudFormation stack "eksctl-multitenant-nodegroup-ng-e4cc9867"
```
### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/defaults.go
Comment (Himangini) [inline pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: Can you add a test for this? Also, people add the output of your manual testing in the PR description

Comment (faraazkhan) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: added unit test for case normalization

Comment (faraazkhan) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: @Himangini please let me know if we need anything else before we can merge this.

Comment (Himangini) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: Can you update your branch and check `Allow edits and access to secrets by maintainers`?

Comment (faraazkhan) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: I have updated the branch, but for some reason I don't see the option to allow edits/access to secrets. 
Screen Shot 2022-10-03 at 1 47 41 PM

**Review by Himangini — APPROVED** [review]
👍🏻

File: pkg/apis/eksctl.io/v1alpha5/defaults.go
Comment (Himangini) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: this is perhaps due to your main branch. Please see the guidelines here https://github.com/weaveworks/eksctl/blob/main/CONTRIBUTING.md#pr-submission-guidelines

Comment (faraazkhan) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: I have attempted creating the PR from a different (not main) branch as well, and it still does not provide the option. Would you like me to close this PR and start a new one from a different branch?

Comment (Himangini) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:170]: I'll make an exception this time, please create a branch for your future contributions 👍🏻

**Review by Himangini — DISMISSED** [review]
looks OK to me.

**Comment by Himangini** [comment]
@faraazkhan are you able to update the branch and merge this?.

**Comment by faraazkhan** [comment]
> @faraazkhan are you able to update the branch and merge this?.

I have updated the branch, but I dont have access to merge. 
967D06BF-33DE-403A-9B17-E8105F8ADBEB

**Comment by Himangini** [comment]
> > @faraazkhan are you able to update the branch and merge this?.
> 
> I have updated the branch, but I dont have access to merge. 967D06BF-33DE-403A-9B17-E8105F8ADBEB

I am unable to merge this, you probably need to close this PR and open it on a forked branch not `main` and enable edits for maintainers 👍🏻

**Comment by faraazkhan** [comment]
> > > > @faraazkhan are you able to update the branch and merge this?.
> > > 
> > > 
> > > I have updated the branch, but I dont have access to merge. !
> > 
> > 
> > I am unable to merge this, you probably need to close this PR and open it on a forked branch not `main` and enable edits for maintainers 👍🏻
> 
> @faraazkhan are you still working on this?

Opened https://github.com/weaveworks/eksctl/pull/5852 from a different branch, closing this.

**Pull request closed without merging**
